### DISCUSSION TITLE
How to run a simple runtime source code with function and call with argument?

### DISCUSSION THREAD

**Discussion by lattice0** [RustPython/RustPython#5173] [Q&A]
```
def hello(s):
    print(s)
```

How do I add this function or module to the VM and call it with an argument?

I found https://github.com/RustPython/RustPython/blob/main/examples/mini_repl.rs#L16 but this is only for compile time python programs. Mine is runtime. I tried all possible methods of the vm, tried setting attributes, etc. I can run source code but cannot call functions inside of it, with arguments, etc.

**Comment by youknowone** [comment]
if the source is file,
```rust
let module = vm.import(<path>)?;
let hello - module.get_attr("hello")?;
vm.invoke(hello, vm.new_pyobj("string"))?;
```

**Reply by lattice0** [reply]
yes. Unfortunately I need it to be just a string and not save it anywhere. Context: a simple web editor that runs the given code, but the code entrypoint is a custom function with custom arguments provided by the web editor server
